core/vm/evm.go

```diff
@@ -419,20 +414,29 @@ func (c *codeAndHash) Hash() common.Hash {
 }
 
 // create creates a new contract using code as deployment code.
-func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64, value *big.Int, address common.Address, typ OpCode) ([]byte, common.Address, uint64, error) {
+func (evm *EVM) create(
+	caller ContractRef,
+	codeAndHash *codeAndHash,
+	gas uint64,
+	value *big.Int,
+	address common.Address,
+	typ OpCode,
+) ([]byte, common.Address, uint64, error) {
 	// Depth check execution. Fail if we're trying to execute above the
 	// limit.
 	if evm.depth > int(params.CallCreateDepth) {
 		return nil, common.Address{}, gas, ErrDepth
 	}
-	if !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
+	if value.Sign() > 0 && !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
```

Comment:
Is this and the check below needed for the objective of this PR, to make precompiles configurable?
If so, why is it needed? Maybe leave a comment explaining why here and below.
If not, maybe move these changes to a separate PR.

Reply:
Mostly added for optimization given we do a lot of deployment (for COA's) without any balance, but will move it to another PR